src/Desktop.Analyzers/UnitTests/DoNotUseInsecureDTDProcessingUseXmlReaderForSchemaReadTests.cs

```diff
@@ -0,0 +1,410 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.UnitTests;
+using Microsoft.CodeAnalysis.CSharp;
+using Xunit;
+
+namespace Desktop.Analyzers.UnitTests
+{
+    public partial class DoNotUseInsecureDTDProcessingAnalyzerTests : DiagnosticAnalyzerTestBase
+    {
+        private DiagnosticResult GetCA3075CSharpResultAt(int line, int column, string name)
+        {
+            return GetCSharpResultAt(line, column, CA3075RuleId, string.Format(CA3075LoadXmlMessage, name, "Read"));
+        }
+
+        private DiagnosticResult GetCA3075BasicResultAt(int line, int column, string name)
+        {
+            return GetBasicResultAt(line, column, CA3075RuleId, string.Format(CA3075LoadXmlMessage, name, "Read"));
+        }
+
+        [Fact]
+        public void UseXmlSchemaReadShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.IO;
+using System.Xml;
+using System.Xml.Schema;
+
+namespace FxCopUnsafeXml
+{
+    public class UseXmlReaderForSchemaRead
+    {
+        public void TestMethod(string path)
+        {
+            TextReader tr = new StreamReader(path);
+            XmlSchema schema = XmlSchema.Read(tr, null);
+        }
+    }
+}",
+                GetCA3075CSharpResultAt(13, 32, "TestMethod")
+            );
+
+            VerifyBasic(@"
+Imports System.IO
+Imports System.Xml
+Imports System.Xml.Schema
+
+Namespace FxCopUnsafeXml
+    Public Class UseXmlReaderForSchemaRead
+        Public Sub TestMethod(path As String)
+            Dim tr As TextReader = New StreamReader(path)
+            Dim schema As XmlSchema = XmlSchema.Read(tr, Nothing)
+        End Sub
+    End Class
+End Namespace",
+                GetCA3075BasicResultAt(10, 39, "TestMethod")
+            );
+        }
+
+        [Fact]
+        public void UseXmlSchemaReadInGetShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.IO;
+using System.Xml.Schema;
```

Comment:
If you don't need `System.Xml` in this test method, I don't think you need it in the one above.